execution/contexts/bond_context.go

```diff
@@ -33,6 +34,11 @@ func (ctx *BondContext) Execute(txe *exec.TxExecution, p payload.Payload) error
 		return err
 	}
 
+	ct := account.PublicKey.GetCurveType()
+	if ct == crypto.CurveTypeSecp256k1 {
```

Comment:
```suggestion
	if ct != crypto.CurveTypeEd25519 {
```
Probably preferred... then again, if tendermint did add, say BLS then it probably would be supported